pdfiumandroid/src/main/cpp/pdfiumandroid.cpp

```diff
@@ -211,9 +248,35 @@ jlong loadTextPageInternal(JNIEnv *env, DocumentFile *doc, jlong pagePtr) {
     }
 }
 
+jfieldID dataBuffer;
+jmethodID readMethod;
+
+extern "C"
+JNIEXPORT jint JNI_OnLoad(JavaVM* vm, void* reserved) {
+    javaVm = vm;
+
+    JNIEnv* env;
+    if (vm->GetEnv(reinterpret_cast<void**>(&env), JNI_VERSION_1_6) != JNI_OK) {
+        return JNI_ERR;
+    }
+
+    jclass pdfiumReaderClass = env->FindClass("io/legere/pdfiumandroid/util/PdfiumNativeSourceBridge");
```

Comment:
Calling methods from JNI is kind similar to calling methods using reflection. We need to find class, find the method/field references and have them ready. A common way to do that is to use method `JNI_OnLoad`: JNI calls it the first time the .so library gets loaded.

First, we store the reference to javaVM to use it later for attaching/detaching the thread. The class is not store anywhere and only used inside the method. methodId and fieldId can be safely stored outside without need for reference wrapping.